pkg/fakerp/store/store.go

```diff
@@ -0,0 +1,123 @@
+package store
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+)
+
+type Store interface {
+	Put(key string, b []byte) error
+	Get(key string) ([]byte, error)
+	Delete(key string) error
+}
+
+type Storage struct {
+	mutex   sync.Mutex
+	mutexes map[string]*sync.Mutex
+	dir     string
+	log     *logrus.Entry
+}
+
+var _ Store = &Storage{}
+
+func New(log *logrus.Entry, dir string) (*Storage, error) {
+	dir = filepath.Clean(dir)
+
+	s := &Storage{
+		dir:     dir,
+		log:     log,
+		mutexes: make(map[string]*sync.Mutex),
+	}
+
+	// if the database already exists, just use it
+	if _, err := os.Stat(dir); err == nil {
+		s.log.Debugf("Using '%s' (database already exists)", dir)
+		return s, nil
+	}
+
+	// if the database doesn't exist create it
+	s.log.Debugf("Creating database at '%s'", dir)
+	return s, os.MkdirAll(dir, 0755)
+}
+
+func (s *Storage) Put(key string, b []byte) error {
+	if key == "" {
+		return fmt.Errorf("missing key - unable to save")
+	}
+
+	mutex := s.getMutex(key)
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	fnlPath := filepath.Join(s.dir, key)
+	tmpPath := fnlPath + ".tmp"
+
+	if err := os.MkdirAll(s.dir, 0755); err != nil {
+		return err
+	}
+
+	if err := ioutil.WriteFile(tmpPath, b, 0644); err != nil {
+		return err
+	}
+
+	return os.Rename(tmpPath, fnlPath)
+}
+
+// Get a record from the database
+func (s *Storage) Get(key string) ([]byte, error) {
```

Comment:
why not work with *api.OpenShiftManagedCluster instead of []byte?

Reply:
It makes the store interface more universal. But :man_shrugging: